src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -34,8 +32,8 @@ RandomFourierGaussPreproc::~RandomFourierGaussPreproc()
 void RandomFourierGaussPreproc::init_basis(int32_t dim_input_space)
 {
 	io::info("Creating Fourier Basis Matrix {}x{}", m_dim_output, dim_input_space);
-	
-	m_basis = sample_spectral_density(dim_input_space);
+
```

Comment:
need a require here that the kernel is definitely set!

Reply:
No, we'd put all the param checks in the fit method. Which I have.

Reply:
put into the docstring of this method that it assumes the kernel is set

Reply:
Noted, we've also to document sampling method. Let's save 'em for later?

Reply:
I think the name is pretty self explaining tbh ... not sure what a docstring would add here